File: services/api-intermediaria/src/main/java/com/gumeinteligencia/api_intermidiaria/entrypoint/mapper/MensagemMapper.java
package com.gumeinteligencia.api_intermidiaria.entrypoint.mapper;

import com.gumeinteligencia.api_intermidiaria.domain.Mensagem;
import com.gumeinteligencia.api_intermidiaria.entrypoint.controller.dto.MensagemDto;

public class MensagemMapper {

    public static Mensagem paraDomain(MensagemDto dto) {
        return Mensagem.builder()
                .telefone(dto.getPhone())
                .mensagem(dto.getText() == null ? "" : dto.getText().getMessage())
                .urlAudio(dto.getAudio() == null ? "" : dto.getAudio().getAudioUrl())
                .urlVideo(dto.getVideo() == null ? "" : dto.getVideo().getVideoUrl())
                .urlImagem(dto.getImage() == null ? "" : dto.getImage().getImageUrl())
                .build();
    }
}

File: services/api-agente/src/entrypoint/mapper/mensagem_mapper.py
from src.domain.mensagem import Mensagem
from src.entrypoint.controller.dto.mensagem_dto import MensagemDto


class MensagemMapper:
    
    def paraDomain(self, dto: MensagemDto) -> Mensagem:
        return Mensagem(
            cliente_id=dto.cliente_id,
            conversa_id=dto.conversa_id,
            message=dto.message,
            audios_url=dto.audios_url,
            imagens_url=dto.imagens_url
        )
        
    def paraDto(self, domain: Mensagem) -> MensagemDto:
        return MensagemDto(
            cliente_id=domain.cliente_id,
            conversa_id=domain.conversa_id,
            message=domain.message,
            audios_url=domain.audios_url,
            imagens_url=domain.imagens_url
        )
